source/Engine.cpp

```diff
@@ -157,22 +157,37 @@ namespace {
 	}
 
 	void DrawFlareSprites(const Ship &ship, DrawList &draw, const vector<Ship::EnginePoint> &enginePoints,
-		const vector<pair<Body, int>> &flareSprites, uint8_t side)
+		const vector<pair<Body, int>> &flareSprites, uint8_t side, Point scale)
 	{
 		for(const Ship::EnginePoint &point : enginePoints)
 		{
 			Point pos = ship.Facing().Rotate(point) * ship.Zoom() + ship.Position();
 			// If multiple engines with the same flare are installed, draw up to
 			// three copies of the flare sprite.
 			for(const auto &it : flareSprites)
-				if(point.side == side && (point.steering == Ship::EnginePoint::NONE
-					|| (point.steering == Ship::EnginePoint::LEFT && ship.SteeringDirection() < 0.)
-					|| (point.steering == Ship::EnginePoint::RIGHT && ship.SteeringDirection() > 0.)))
-					for(int i = 0; i < it.second && i < 3; ++i)
-					{
-						Body sprite(it.first, pos, ship.Velocity(), ship.Facing() + point.facing, point.zoom);
-						draw.Add(sprite, ship.Cloaking());
-					}
+				if(point.side == side)
+				{
+					if(point.steering == Ship::EnginePoint::NONE)
+						for(int i = 0; i < it.second && i < 3; ++i)
+						{
+							Body sprite(it.first, pos, ship.Velocity(), ship.Facing() + point.facing, point.zoom, scale);
+							draw.Add(sprite, ship.Cloaking());
+						}
+					else if(point.steering == Ship::EnginePoint::LEFT && ship.TurnLeftHeldFrames())
+						for(int i = 0; i < it.second && i < 3; ++i)
+						{
+							Body sprite(it.first, pos, ship.Velocity(), ship.Facing() + point.facing, point.zoom,
+								scale * Point(ship.TurnLeftHeldFrames() / 10., pow(ship.TurnLeftHeldFrames() / 10., 3)));
```

Comment:
this part here seems to be the only difference so mb we could set it in a variable and keep doing it all in the same place; with that one variable changing?